File: Dsa-day91.py
class Solution:
    def findSmallestInteger(self, nums: List[int], value: int) -> int:
        freq = Counter(x % value for x in nums)
        i = 0
        while True:
            if freq[i % value] == 0:
                return i
            freq[i % value] -= 1
            i += 1

File: Dsa-day91.java
class Solution {
    public int findSmallestInteger(int[] nums, int value) {
        HashMap<Integer, Integer> freq = new HashMap<>();
        for (int x : nums) {
            int r = ((x % value) + value) % value;
            freq.put(r, freq.getOrDefault(r, 0) + 1);
        }
        int i = 0;
        while (true) {
            int r = i % value;
            if (!freq.containsKey(r) || freq.get(r) == 0) return i;
            freq.put(r, freq.get(r) - 1);
            i++;
        }
    }
}
